src/Agent.Listener/Configuration/AutoLogonManager.cs

```diff
@@ -0,0 +1,198 @@
+            Trace.Info("Successfully deleted the autologon settings.");
+        }
+
+        private void SaveAutoLogonSettings(string domainName, string userName)
+        {
+            Trace.Entering();
+            var settings = new AutoLogonSettings()
+            {
+                UserDomainName = domainName,
+                UserName = userName
+            };            
+            _store.SaveAutoLogonSettings(settings);
+            Trace.Info("Saved the autologon settings");
+        }
+
+        private async Task ConfigurePowerOptions()
+        {
+            var whichUtil = HostContext.GetService<IWhichUtil>();
+            var filePath = whichUtil.Which("powercfg.exe", require:true);
+            string[] commands = new string[] {"/Change monitor-timeout-ac 0", "/Change monitor-timeout-dc 0"};
+
+            foreach (var command in commands)
+            {
+                try
+                {
+                    Trace.Info($"Running powercfg.exe with {command}");
+                    using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                    {
+                        processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                        {
+                            Trace.Info(message.Data);
+                        };
+
+                        processInvoker.ErrorDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                        {
+                            Trace.Error(message.Data);
+                            _terminal.WriteError(message.Data);
+                        };
+
+                        await processInvoker.ExecuteAsync(
+                                workingDirectory: string.Empty,
+                                fileName: filePath,
+                                arguments: command,
+                                environment: null,
+                                cancellationToken: CancellationToken.None);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //we will not stop the configuration. just show the warning and continue
+                    _terminal.WriteError(StringUtil.Loc("PowerOptionsConfigError"));
+                    Trace.Error(ex);
+                }
+            }
+        }
+
+        private void RestartBasedOnUserInput(CommandSettings command)
+        {
+            Trace.Info("Asking the user to restart the machine to launch agent and for autologon settings to take effect.");
+            _terminal.WriteLine(StringUtil.Loc("RestartMessage"));
+
+            var shallRestart = command.GetRestartNow();
+            if (shallRestart)
+            {
+                var whichUtil = HostContext.GetService<IWhichUtil>();
+                var shutdownExePath = whichUtil.Which("shutdown.exe");
+
+                Trace.Info("Restarting the machine in 5 seconds");
```

Comment:
should we increase the time? 5 seconds is too short for missing click restart. might be better for 15s~30s?